### DISCUSSION TITLE
Transpiler itu apa?

### DISCUSSION THREAD

**Discussion by giscus** [r17x/rin.rocks#34] [Q&A]
# Transpiler itu apa?

Apakah kamu pernah berfikir untuk memahami lebih dalam bagaimana sebuah bahasa pemrograman itu dimaknai (berjalan) ?

https://www.rin.rocks/posts/transpeler-atau-compiler

**Comment by fzn0x** [comment]
Sebelum membaca postingan ini dan cuman baca wikipedia:

Transpiler = source-to-source
Compiler = high level-to-low level

Setelah membaca postingan ini dan riset lagi:

Transpeler = abstraction-to-abstraction
Compiler = level-to-level karena self hosted compiler -> Rust compiler ditulis di Rust untuk menjalankan Rust  (https://github.com/rust-lang/rust)

Thank you bang!

**Reply by r17x** [reply]
Rata - rata compiler bakalan ditulis oleh diri mereka sendir. dulu rust dibberapa bagian compilernya masih menggunakan OCaml sebelum dia "self hosted". @fzn0x
